secboot/secboot_dummy.go

```diff
@@ -98,3 +98,5 @@ func GetPrimaryKeyHMAC(devicePath string, alg crypto.Hash) ([]byte, []byte, erro
 func VerifyPrimaryKeyHMAC(devicePath string, alg crypto.Hash, salt []byte, digest []byte) (bool, error) {
 	return false, errBuildWithoutSecboot
 }
+
+type HashAlg crypto.Hash
```

Comment:
does this need marshal/unmarshal json implementation?

Reply:
Yes, but with nosecboot, we do not care. It will just serialize to `{}`. And that is that.

Reply:
I suppose we should at least keep the JSON type consistent then for both cases. IIRC secboot serializes to string, so this should be sufficient:

```
// stay consistent with secboot serialization of HashAlg
func (h *HashAlg) MarshalJSON() ([]byte, error) { return []byte(`""`), nil } 
func (h *HashAlg) UnmarshalJSON([]byte) error { *h = nil ; return nil } 
```

Reply:
So that it marshals to `""` instead of `{}`? Is there a point? It is just json either way.

Reply:
I think we should implement `MarshalJSON` and `UnmarshalJSON`, we should just return error that it is not implemented instead. It should be an error that we try to serialize a state in nosecboot.

Reply:
From the discusion on MM, this could cause failure in as early as fde manager startup, but since the startup errors aren't fatal now, it is theoretically safe. So, the remaining question is whether we can rely on backward compatibility in secboot's own marshaller.

Reply:
I had to revert that because unit tests did not work. If we want that, we need to disable fdemgr unit tests on nosecboot.